docs/release.md

```diff
@@ -17,43 +17,34 @@ Non-breaking changes are typically merged very quickly and often released as pat
 
 Every crate that we publish on `crates.io` has a `CHANGELOG.md` file.
 Substantial PRs should add an entry to each crate they modify.
-The next unreleased version is tagged with ` - unreleased`, for example: `0.17.0 - unreleased`.
+We have a CI check[^1] that enforces adding a changelog entry if you modify code in a particular crate.
+In case the current version is already released (we also check that in CI), you'll have to add a new header at the top.
```

Comment:
This in particular makes it a bit harder to contribute. Previously it was clear from changelog alone if it is released or not. Now I guess I'll need to check crates.io as well to see if I need to add new section or not.

This whole changelog management process seems quite heavy to be honest.

Reply:
I recognize that it is not ideal but I'd assume that most PRs go through at least one review cycle and multiple CI runs. Thus, a viable approach for contributors should be to just add to the top-most changelog entry and see if CI accepts that or not.

In fact, other than for experienced contributors like yourself, I'd expect this to be the default.

> This whole changelog management process seems quite heavy to be honest.

I am definitely open to other ideas. So far, I've mostly tried to optimise for correctness and ease of maintainability, hence the high degree of automation here.

Reply:
If checks are smart enough to verify that already released entry is updated with more changes I think it is good, can't imagine how much debugging it took to make it work :open_mouth:

Reply:
> If checks are smart enough to verify that already released entry is updated with more changes I think it is good, can't imagine how much debugging it took to make it work 😮

Essentially, we check `top_most_version_in_changelog_is_unreleased && changelog_has_been_modified`. There is a corner-case where the top-most entry is unreleased and you modify an entry further down below. I don't really see why anybody would do that so I didn't bother codifying that :)

It took several attempts at making the checks robust but they've been working quite well recently :)